ui: clusterviz: onboarding instructions box

vilterp opened the pull request:
Expanded state:
![image](https://user-images.githubusercontent.com/7341/37975492-7479c8b2-31ad-11e8-95f4-4cf3f686831c.png)
Collapsed state:
![image](https://user-images.githubusercontent.com/7341/37975518-87085638-31ad-11e8-8bd0-2a5e4845f437.png)

Also link to setup guide from enterprise gate page:
![image](https://user-images.githubusercontent.com/7341/37975790-1824e320-31ae-11e8-93f9-db9fbefdbf08.png)

Commit:
ui: clusterviz: add box with screenshot config instructions

The box appears when
- you are looking at the circle layout
- AND the circle layout has some localities in it
- AND some nodes don't have localities.

Release note (admin ui change): Add box in bottom right of cluster viz
with instructions on how to configure it to get a map.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/23516)
<!-- Reviewable:end -->

couchand commented on pkg/ui/src/util/locations.ts:
```diff
@@ -89,3 +89,44 @@ export function findOrCalculateLocation(locations: LocationTree, locality: Local
+/*
+ * getConfigStats finds how many nodes are missing locality or location config.
+ */
+export function getConfigStatus(
```
can you add tests for this method please?

vilterp replied:
Removed this method.

couchand commented on pkg/ui/src/util/locations.ts:
```diff
@@ -89,3 +89,44 @@ export function findOrCalculateLocation(locations: LocationTree, locality: Local
+  let totalNodes = 0;
+  let withLocality = 0;
+  let withLocation = 0;
+  function recur(tree: LocalityTree) {
```
it may be a moot point, but I bet this method would be much clearer if it just operated on the list of nodes, rather than the locality tree.

vilterp replied:
Good call. This function used to do more, but now the recursion is completely unnecessary.

couchand commented on pkg/ui/src/util/locations.ts:
```diff
@@ -89,3 +89,44 @@ export function findOrCalculateLocation(locations: LocationTree, locality: Local
+  function recur(tree: LocalityTree) {
+    tree.nodes.forEach(() => {
+      totalNodes++;
+      // TODO: what about root?
```
what about root?

couchand commented on pkg/ui/src/util/locations.ts:
```diff
@@ -89,3 +89,44 @@ export function findOrCalculateLocation(locations: LocationTree, locality: Local
+        withLocality++;
+        const currentTier = tree.tiers[tree.tiers.length - 1];
+        if (hasLocation(locationTree, currentTier)) {
+          withLocation++;
```
yes, as you TODO'd above, localities have locations, not nodes.  it's not exactly clear how to specify "how many localities are left" in the general case.  perhaps rather than trying to be this specific, we should provide more general feedback and a link to the debug page?

vilterp replied:
Done.

couchand commented on pkg/ui/ccl/src/views/clusterviz/containers/map/nodeCanvas.tsx:
```diff
@@ -22,6 +22,8 @@ import { generateLocalityRoute, getLocalityLabel } from "src/util/localities";
 import { trustIcon } from "src/util/trust";
 import { cockroach } from "src/js/protos";
+import {getConfigStatus} from "oss/src/util/locations";
+import {InstructionsBox} from "oss/ccl/src/views/clusterviz/components/instructionsBox";
```
oss/ccl

vilterp replied:
Done.

couchand commented on pkg/ui/ccl/src/views/clusterviz/containers/map/nodeCanvas.tsx:
```diff
@@ -153,10 +166,11 @@ export class NodeCanvas extends React.Component<NodeCanvasProps, NodeCanvasState
           </svg>
         </div>
         { this.renderBackButton() }
+        { showMap ? null : this.renderInstructionsBox() }
```
i don't think that logic is right

couchand commented on pkg/ui/ccl/src/views/clusterviz/containers/map/nodeCanvas.tsx:
```diff
@@ -71,15 +73,15 @@ export class NodeCanvas extends React.Component<NodeCanvasProps, NodeCanvasState
   }
 
-  renderContent() {
+  renderContent(asMap: boolean) {
```
well that's an unpleasant change.  boolean parameters are bad.  perhaps just move the conditional up to the parent?

vilterp replied:
Haven't seen an elegant way to get rid of this. Let's look at it together.

couchand commented on pkg/ui/ccl/src/views/clusterviz/containers/map/nodeCanvas.tsx:
```diff
@@ -136,7 +138,18 @@ export class NodeCanvas extends React.Component<NodeCanvasProps, NodeCanvasState
     );
   }
 
+  renderInstructionsBox() {
```
given how little content this has, the method doesn't seem to add much value

vilterp replied:
Done.

couchand commented on pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx:
```diff
@@ -0,0 +1,91 @@
+  }
+
+  renderCollapsed() {
+    return "?";
```
hmm... i think the dismissal should actually dismiss.  isn't "i don't want a map, get outta my face" a valid use-case?

vilterp replied:
Collapses to a very small and unobtrusive question mark.

couchand commented on pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx:
```diff
@@ -0,0 +1,91 @@
+export class InstructionsBox extends React.Component<InstructionsBoxProps, InstructionBoxState> {
+  constructor() {
+    super();
+    const valFromStorage = localStorage.getItem(LOCAL_STORAGE_KEY);
```
it seems like it would be nice to have this use the same mechanism that the version dismissal uses, for consistency.  it doesn't look that complicated, take a look at `src/redux/alerts`

couchand commented on pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx:
```diff
@@ -0,0 +1,91 @@
+    return (
+      <div style={{ maxWidth: 220 }}>
+        <p>
+          To see a map, {todos.join(" and ")}.{" "}
```
am i missing where this goes away if `todos.length === 0`?

vilterp replied:
Done.

vilterp replied:
This component won't even render.

couchand commented on pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx:
```diff
@@ -0,0 +1,91 @@
+    }
+
+    return (
+      <div onClick={() => this.handleToggle()} className="instructions-box">
```
please autobind so we don't make a new closure on each render

vilterp replied:
Done.

vilterp replied:
Done.

vilterp commented:
Design notes:
- more padding inside pip
- bold stuff

vilterp commented on pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx:
```diff
@@ -0,0 +1,132 @@
+  setExpanded: (expanded: boolean) => void;
+}
+
+const DOCS_LINK = "http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/50893285dcb2e74c603d34103597b83f715a4c31/dev/admin-ui-node-map.html#configure-and-navigate-the-node-map";
```
TODO: get published docs link before merging

vilterp commented:
@Amruta-Ranade this page introduces a link to the node map docs, and we'd like to merge it in the next day or two. Not sure if you will merge your docs by then. Should I merge this with the link that the node map docs _will be_ at? (I think you told me this URL once; I can't find it though)

Amruta-Ranade commented:
I don't think the PR will be merged so soon. Here's the (most probable) link: https://www.cockroachlabs.com/docs/dev/admin-ui-node-map.html 
(I need to check what happens when we make 2.0 docs the stable ones instead of the dev ones. Ping me again before merging your PR?)

vilterp commented:
Maybe we should use one of those fancy schmancy bit.ly links (`cockroa.ch`) so we can control the redirect later, and get tracking. Thoughts, @couchand? I think that's the only issue blocking this, other than what you find in your review.

Amruta-Ranade commented:
AFAIK, they don't let you redirect links in the free version of the product. Let me ask Jesse if we can redirect the docs page links.

couchand commented:
@vilterp use the helper `docsURL` from `src/util/docs` to link directly to the version of docs corresponding to the binary, rather than dev/stable.  We should just pick a page name and commit to it.  It will be a broken link until the docs are merged, but I don't think there's a way around that (other than linking to a TC build, which seems really weird).  Then we should just lend a hand to help @Amruta-Ranade get the PR across the finish line there.

couchand commented:
For the most part looks fine.  I left a number of comments that can probably be todo-later, but a some of the big ones seem to be worth addressing now.

---

Reviewed 2 of 8 files at r5, 5 of 9 files at r7, 6 of 6 files at r8.
Review status: all files reviewed at latest revision, 19 unresolved discussions, some commit checks failed.

---

*[pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx, line 51 at r8](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8DMBAEvkZNhH9PbTUN:-L8DMBAEvkZNhH9PbTUO:b-exi5ch) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx#L51)):*
> ```XML
>           <span
>             className="instructions-box-top-bar__x_out"
>             onClick={() => this.props.setExpanded(false)}
> ```

please prebind

---

*[pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx, line 73 at r8](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8DMKU-nPZaPRh73MM6:-L8DMKU-nPZaPRh73MM7:b-chs1z4) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx#L73)):*
> ```XML
>       <div
>         className="instructions-box instructions-box--collapsed"
>         onClick={() => this.props.setExpanded(true)}
> ```

likewise prebind

---

*[pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx, line 92 at r8](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8DNFeFT99_v8qgtQ5e:-L8DNFeFT99_v8qgtQ5f:b-vi9n76) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx#L92)):*
> ```XML
>   return {
>     expanded: instructionsBoxExpandedSelector(state),
>     allNodes: nodeStatusesSelector(state),
> ```

maybe move the `numNodesWithoutLocality` call here?

---

*[pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx, line 103 at r8](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8DKm885S4ijVUs39lK:-L8DKm885S4ijVUs39lL:b-fy5t9h) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx#L103)):*
> ```XML
>         key: INSTRUCTIONS_BOX_EXPANDED_KEY,
>         value: expanded,
>       }));
> ```

it seems like it would be nice to put the above next to the rest of the settings handling code, perhaps as a thunk action creator?

---

*[pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx, line 119 at r8](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8D37css1ubRAN6Xs1F:-L8D37css1ubRAN6Xs1G:b-107nej) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx#L119)):*
> ```XML
> /**
>  * showInstructionBox decides whether to show the instructionBox.
>  * Exported for testing.
> ```

And also the actual usage...

---

*[pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx, line 125 at r8](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8DLJ7fNDJzRSz-iQ-W:-L8DLJ7gS3RytILPgl6r:blxiq24) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx#L125)):*
> ```XML
>  * It's also false if we're down a level and looking at all nodes.
>  */
> export function showInstructionsBox(
> ```

This logic is fine, though I do think it would be cleaner to have it rely purely on the state: the node's locality flags and the locations table should be sufficient.  Perhaps we leave that for future work...

---

*[pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx, line 132 at r8](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8DG0IPoWvRwRNDCqd1:-L8DG0IQ0pYOJY2BEk7R:b-k9crio) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx#L132)):*
> ```XML
>   }
>   const downALevel = tiers.length > 0;
>   const justNodes = _.size(localityTree.localities) === 0;
> ```

maybe `_.isEmpty`?

---

*[pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx, line 141 at r8](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8DEdmLF8oM4JkeCi6H:-L8DEdmLF8oM4JkeCi6I:bnli7pn) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx#L141)):*
> ```XML
>   if (nodesWithoutLocality > 0) {
>     return {
>       num: 1,
> ```

Do these step numbers correspond with anything else?  I think it may be confusing if you only see one of them at a time and there's not more context.

---

*[pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx, line 146 at r8](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8DEsA4MQFJ2CiVPO2u:-L8DEsA4MQFJ2CiVPO2v:b-23j0v8) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx#L146)):*
> ```XML
>           Start every node in the cluster with a <code>--locality</code> flag.
>           <br />
>           ({nodesWithoutLocality} currently missing it)
> ```

I'm not sure the count is all that helpful...  At least without it we can plausibly claim that this is "static", but with the count there it just calls attention to the fact that we don't provide the answer to the question "which ones?" and we're probably not going to get to answering that question right now...

---

*[pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx, line 156 at r8](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8DIfGVmxhXbY1t2rh8:-L8DIfGWpwd8CqX-uc1m:b-feiujm) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx#L156)):*
> ```XML
>       <span>
>         Add locations to the <code>system.locations</code> table corresponding to
>         your localities.
> ```

This copy could use clarification

---

*[pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/instructionsBox.styl, line 9 at r8](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8DDE7FXhFy38S91I13:-L8DDE7FXhFy38S91I14:b-w98c2v) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/instructionsBox.styl#L9)):*
> ```Stylus
>   top 26px
>   background-color white
>   border 1px solid #EDEDED
> ```

can some/all of these colors reference `styl/base/palette` please?

---

*[pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/instructionsBox.styl, line 23 at r8](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8DDbiTUk3u5WqldEB3:-L8DDbiTUk3u5WqldEB4:b-b41hz4) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/instructionsBox.styl#L23)):*
> ```Stylus
>     color #3A7DE1
> 
>   &__collapse_icon
> ```

it looks like some of these styles are unused?  can you go through and make sure they're cleaned up?

---

*[pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/instructionsBox.styl, line 64 at r8](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8DDt45-ScfwY8LQJ5O:-L8DDt45-ScfwY8LQJ5P:boqvzsl) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/instructionsBox.styl#L64)):*
> ```Stylus
> 
>   &__screenshot
>     // Crop the image.
> ```

could we just crop it in gimp or something instead?

---

*[pkg/ui/ccl/src/views/clusterviz/containers/map/nodeCanvas.tsx, line 163 at r8](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8DAK4i3s215q8Ql7XX:-L8DAK4i3s215q8Ql7XY:bgcab2r) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/ccl/src/views/clusterviz/containers/map/nodeCanvas.tsx#L163)):*
> ```XML
>         </div>
>         { this.renderBackButton() }
>         { showInstructionsBox(showMap, this.props.tiers, this.props.localityTree)
> ```

It should be the case that `this.props.tiers` is the same as `this.props.localityTree.tiers`, so now I'm trying to remember why both are props here.  I think, at least for your usage, `showInstructionsBox` could get the tiers off the locality tree to eliminate the ambiguity.

---

*[pkg/ui/src/redux/alerts.spec.ts, line 417 at r8](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8DC2RIwUETbAWlmZK4:-L8DC2RIwUETbAWlmZK5:b-19lihv) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/src/redux/alerts.spec.ts#L417)):*
> ```XML
>       })));
>       dispatch(setUIDataKey(VERSION_DISMISSED_KEY, "blank"));
>       dispatch(setUIDataKey(INSTRUCTIONS_BOX_EXPANDED_KEY, false));
> ```

I'm having a hard time figuring out what precisely this test is verifying, but it seems like testing the logic you've written would be a reasonable idea?  Perhaps as future work...

---

*[pkg/ui/src/redux/alerts.ts, line 46 at r8](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8D7D_VxTTSOCkRAgC3:-L8D7D_VxTTSOCkRAgC4:bdccjkf) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/src/redux/alerts.ts#L46)):*
> ```XML
> const localSettingsSelector = (state: AdminUIState) => state.localSettings;
> 
> // Clusterviz Instruction Box collapsed
> ```

This comment doesn't match the implementation.  In a case like this, I actually prefer the comment's logic: it seems natural that having never interacted with it would use a falsey value, so that `null`/`undefined` work just as well as `false`.

---

*[pkg/ui/src/redux/alerts.ts, line 57 at r8](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8D7uxfg_NumPK41yCc:-L8D7uxfg_NumPK41yCd:b-pff3en) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/src/redux/alerts.ts#L57)):*
> ```XML
>     uiData
>       && _.has(uiData, INSTRUCTIONS_BOX_EXPANDED_KEY)
>       && uiData[INSTRUCTIONS_BOX_EXPANDED_KEY].status === UIDataStatus.VALID
> ```

does the fact that this condition is missing from `newVersionDismissedPersistentLoadedSelector` (btw what a mouthful) indicate a bug there?  is this how you addressed the optimistic update?

---

*[pkg/ui/src/util/localities.ts, line 105 at r8](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8D4rV6oL5_VxV8v-M6:-L8D4rV6oL5_VxV8v-M7:bmpgpqk) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/src/util/localities.ts#L105)):*
> ```XML
>  * numNodesWithoutLocality finds how many nodes are missing locality flags.
>  */
> export function numNodesWithoutLocality(nodes: NodeStatus$Properties[]): number {
> ```

I don't like boolean parameters, but I love functions that return booleans (i.e. predicates).  Since in actual usage all you care about is zero/non-zero, perhaps this could be changed to `anyNodesMissingLocality` or something?

Alternatively, go the other way and make this just `nodesWithoutLocality` and do the filter but not the `.length`?

Of course not necessary for now, but something to think about.

--

Also, test?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-:-L8DOFIELzsUUcnkZtK7:b-ts2ty1)*
<!-- Sent from Reviewable.io -->

Amruta-Ranade commented:
Here's the link to the Docs page: 
https://www.cockroachlabs.com/docs/stable/enable-node-map.html

(Placeholder content for now, but I won't change the link)

vilterp commented:
Review status: all files reviewed at latest revision, 21 unresolved discussions, some commit checks failed.

---

*[pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx, line 51 at r8](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8DMBAEvkZNhH9PbTUN:-L8ECSLovlEI9jlnYpnB:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx#L51)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

please prebind
</blockquote></details>

Done.

---

*[pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx, line 73 at r8](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8DMKU-nPZaPRh73MM6:-L8ECSijVql7hGzgeuGa:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx#L73)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

likewise prebind
</blockquote></details>

Done.

---

*[pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx, line 92 at r8](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8DNFeFT99_v8qgtQ5e:-L8ECYW644cZvetvrIEM:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx#L92)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

maybe move the `numNodesWithoutLocality` call here?
</blockquote></details>

Done.

---

*[pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx, line 103 at r8](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8DKm885S4ijVUs39lK:-L8EIo4Q556dlF9Fug7Z:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx#L103)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

it seems like it would be nice to put the above next to the rest of the settings handling code, perhaps as a thunk action creator?
</blockquote></details>

Done.

---

*[pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx, line 119 at r8](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8D37css1ubRAN6Xs1F:-L8EIoUohi1aMglLLP5C:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx#L119)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

And also the actual usage...
</blockquote></details>

Done.

---

*[pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx, line 125 at r8](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8DLJ7fNDJzRSz-iQ-W:-L8EMjsR8VXos1gfuT6_:bk4uxh1) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx#L125)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

This logic is fine, though I do think it would be cleaner to have it rely purely on the state: the node's locality flags and the locations table should be sufficient.  Perhaps we leave that for future work...
</blockquote></details>

Removed a parameter. What do you mean by "rely purely on state"? What state?

---

*[pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx, line 132 at r8](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8DG0IPoWvRwRNDCqd1:-L8EMev1-lT8BaOwp1tM:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx#L132)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

maybe `_.isEmpty`?
</blockquote></details>

Done.

---

*[pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/instructionsBox.styl, line 23 at r8](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8DDbiTUk3u5WqldEB3:-L8EJTuQOO4CPJH3W-Ra:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/instructionsBox.styl#L23)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

it looks like some of these styles are unused?  can you go through and make sure they're cleaned up?
</blockquote></details>

Done.

---

*[pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/instructionsBox.styl, line 64 at r8](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8DDt45-ScfwY8LQJ5O:-L8EJUyOOWYzvVeLkXMj:bzjwl1) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/instructionsBox.styl#L64)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

could we just crop it in gimp or something instead?
</blockquote></details>

This image is also used in the enterprise gate page (as one of the tree boxes) and needs to be taller there.

---

*[pkg/ui/ccl/src/views/clusterviz/containers/map/nodeCanvas.tsx, line 173 at r4](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L6xXy17O1Hw90h17RST-r4-173:-L8EJhTFJBGyh82Zj-Da:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/5fe4b53d945ce74596b63b601c314069b8078107/pkg/ui/ccl/src/views/clusterviz/containers/map/nodeCanvas.tsx#L173)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

i don't think that logic is right
</blockquote></details>

Done.

---

*[pkg/ui/ccl/src/views/clusterviz/containers/map/nodeCanvas.tsx, line 163 at r8](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8DAK4i3s215q8Ql7XX:-L8EJhifKV-isY6cyqUg:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/ccl/src/views/clusterviz/containers/map/nodeCanvas.tsx#L163)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

It should be the case that `this.props.tiers` is the same as `this.props.localityTree.tiers`, so now I'm trying to remember why both are props here.  I think, at least for your usage, `showInstructionsBox` could get the tiers off the locality tree to eliminate the ambiguity.
</blockquote></details>

Done.

---

*[pkg/ui/src/redux/alerts.spec.ts, line 417 at r8](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8DC2RIwUETbAWlmZK4:-L8EJmC1Yzo7ZBiQbT9w:bkn5nt2) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/src/redux/alerts.spec.ts#L417)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

I'm having a hard time figuring out what precisely this test is verifying, but it seems like testing the logic you've written would be a reasonable idea?  Perhaps as future work...
</blockquote></details>

This is testing the `alertsDataSync` function, which I haven't changed except to add `INSTRUCTIONS_BOX_EXPANDED_KEY` as one of the keys that needs to be saved.

---

*[pkg/ui/src/redux/alerts.ts, line 46 at r8](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8D7D_VxTTSOCkRAgC3:-L8ELtLx0_logaCuJf0V:b-z36hzf) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/src/redux/alerts.ts#L46)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

This comment doesn't match the implementation.  In a case like this, I actually prefer the comment's logic: it seems natural that having never interacted with it would use a falsey value, so that `null`/`undefined` work just as well as `false`.
</blockquote></details>

Not following; I think this comment may have been left on different code than what I see in the diff now…

---

*[pkg/ui/src/redux/alerts.ts, line 57 at r8](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8D7uxfg_NumPK41yCc:-L8ELzC5umPx7iaOlJ22:b-y8d4oc) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/src/redux/alerts.ts#L57)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

does the fact that this condition is missing from `newVersionDismissedPersistentLoadedSelector` (btw what a mouthful) indicate a bug there?  is this how you addressed the optimistic update?
</blockquote></details>

Yeah, this is kind of an insane amount of repeating oneself to store one boolean. But yes, the loaded selector is for optimistic update. Follows the pattern of the "version dismissed" selectors below.

Anyway, added that condition in. But, it shouldn't matter since the result of this selector is disregarded if the setting is not loaded.

---

*[pkg/ui/src/util/locations.ts, line 105 at r4](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L6xXy17O1Hw90h17RSU-r4-105:-L8ECAXiGPknmsPMLtPl:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/5fe4b53d945ce74596b63b601c314069b8078107/pkg/ui/src/util/locations.ts#L105)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

what about root?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516)*
<!-- Sent from Reviewable.io -->

couchand commented:
Reviewed 7 of 7 files at r10.
Review status: all files reviewed at latest revision, 10 unresolved discussions, some commit checks failed.

---

*[pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx, line 125 at r8](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8DLJ7fNDJzRSz-iQ-W:-L8ESRvVl6HT5IuvgHNv:b-8yrwrw) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx#L125)):*
<details><summary><i>Previously, vilterp (Pete Vilter) wrote…</i></summary><blockquote>

Removed a parameter. What do you mean by "rely purely on state"? What state?
</blockquote></details>

`AdminUIState`.  It's basically check it we'd show a map at the root of the locality tree.  But now that we're just showing it at the root, it's more direct.

---

*[pkg/ui/ccl/src/views/clusterviz/containers/map/nodeCanvas.tsx, line 163 at r10](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8ET9UFqoyoGZWJ-XgE:-L8ET9UGJNX3aiZfbB4W:b-kiysuq) ([raw file](https://github.com/cockroachdb/cockroach/blob/fe8a49fe220baaaa73fba9a6c0470599de9c0de9/pkg/ui/ccl/src/views/clusterviz/containers/map/nodeCanvas.tsx#L163)):*
> ```XML
>         </div>
>         { this.renderBackButton() }
>         { showInstructionsBox(showMap, this.props.tiers, this.props.localityTree)
> ```

extra param here now

---

*[pkg/ui/src/redux/alerts.ts, line 46 at r8](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8D7D_VxTTSOCkRAgC3:-L8ETFPyRfSstU2DmmCl:b-7yh9c8) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/src/redux/alerts.ts#L46)):*
<details><summary><i>Previously, vilterp (Pete Vilter) wrote…</i></summary><blockquote>

Not following; I think this comment may have been left on different code than what I see in the diff now…
</blockquote></details>

It looks like you updated the comment to say "expanded".  My point above remains, I think this would be cleaner if the logic were reversed and "collapsed" were the stored value...

---

*[pkg/ui/src/redux/alerts.ts, line 57 at r8](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8D7uxfg_NumPK41yCc:-L8ETcnhr5ZAc5bfWCRD:bbqqin5) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/src/redux/alerts.ts#L57)):*
<details><summary><i>Previously, vilterp (Pete Vilter) wrote…</i></summary><blockquote>

Yeah, this is kind of an insane amount of repeating oneself to store one boolean. But yes, the loaded selector is for optimistic update. Follows the pattern of the "version dismissed" selectors below.

Anyway, added that condition in. But, it shouldn't matter since the result of this selector is disregarded if the setting is not loaded.
</blockquote></details>

Sorry, I meant that here you're checking `status === VALID` but the `newVersionDismissedPersistentLoadedSelector` doesn't -- does it have a bug related to error states?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516)*
<!-- Sent from Reviewable.io -->

Commit:
ui: update enterprise gate page to link to node map docs

Release note (admin ui change): update node map's "you need a license"
page to link to node map setup guide

vilterp commented:
Review status: 2 of 11 files reviewed at latest revision, 10 unresolved discussions.

---

*[pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx, line 141 at r8](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8DEdmLF8oM4JkeCi6H:-L8Ec8S-lBZ5AiTKyEnK:b-nxi1a1) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx#L141)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

Do these step numbers correspond with anything else?  I think it may be confusing if you only see one of them at a time and there's not more context.
</blockquote></details>

Done. Now showing them both.

---

*[pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx, line 146 at r8](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8DEsA4MQFJ2CiVPO2u:-L8EcAldZglYwUueVaqA:bss4a2b) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx#L146)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

I'm not sure the count is all that helpful...  At least without it we can plausibly claim that this is "static", but with the count there it just calls attention to the fact that we don't provide the answer to the question "which ones?" and we're probably not going to get to answering that question right now...
</blockquote></details>

Removed.

---

*[pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx, line 156 at r8](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8DIfGVmxhXbY1t2rh8:-L8EcwSYkquU-ImXmgpI:b-fnryro) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx#L156)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

This copy could use clarification
</blockquote></details>

Updated; I think it's a little better.

---

*[pkg/ui/ccl/src/views/clusterviz/containers/map/nodeCanvas.tsx, line 163 at r10](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8ET9UFqoyoGZWJ-XgE:-L8EczZZxR_C4bcffMxD:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/fe8a49fe220baaaa73fba9a6c0470599de9c0de9/pkg/ui/ccl/src/views/clusterviz/containers/map/nodeCanvas.tsx#L163)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

extra param here now
</blockquote></details>

Done.

---

*[pkg/ui/src/redux/alerts.ts, line 57 at r8](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8D7uxfg_NumPK41yCc:-L8Ed6eAX2Yg8nlJZupJ:bvj4mzx) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/src/redux/alerts.ts#L57)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

Sorry, I meant that here you're checking `status === VALID` but the `newVersionDismissedPersistentLoadedSelector` doesn't -- does it have a bug related to error states?
</blockquote></details>

They both do now

---

*[pkg/ui/src/util/localities.ts, line 105 at r8](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8D4rV6oL5_VxV8v-M6:-L8EfGyc5otd2e-SvHe3:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/src/util/localities.ts#L105)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

I don't like boolean parameters, but I love functions that return booleans (i.e. predicates).  Since in actual usage all you care about is zero/non-zero, perhaps this could be changed to `anyNodesMissingLocality` or something?

Alternatively, go the other way and make this just `nodesWithoutLocality` and do the filter but not the `.length`?

Of course not necessary for now, but something to think about.

--

Also, test?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516)*
<!-- Sent from Reviewable.io -->

Amruta-Ranade commented:
Text and related suggestions:
The header text (don't know what's the technical term, but the one that starts with "Get a clearer view..."
**View the Node Map**
The Node Map shows the geographical layout of your cluster, along with metrics and health indicators. To enable the Node Map, request Enterprise Trial license and refer to this configuration guide (link to the Enable Node Map doc).

Box 1:
The screenshot in the image of the fake trial license: It could be interpreted as if we are okay with them using the license shown. To be safe, we might want to add the screenshot of the Request trial license page and link to it.

Box 3:
Text suggestion: Refer to [Enable Node Map](link to the doc) to configure the Node Map.

Commit:
ui: use palette colors in instructions box and enterprise gate page

Release note: None

vilterp commented:
Giving others a chance to weigh in on the copy; I think I've addressed everything else.

---

Review status: 2 of 11 files reviewed at latest revision, 10 unresolved discussions.

---

*[pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/instructionsBox.styl, line 9 at r8](https://beta.reviewable.io/reviews/cockroachdb/cockroach/23516#-L8DDE7FXhFy38S91I13:-L8F1Edf5NYq_BEOsYjc:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/instructionsBox.styl#L9)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

can some/all of these colors reference `styl/base/palette` please?
</blockquote></details>

Done.

---

*[pkg/ui/src/redux/alerts.ts, line 46 at r8](https://beta.reviewable.io/reviews/cockroachdb/cockroach/23516#-L8D7D_VxTTSOCkRAgC3:-L8F1G_Y4nvzQ_c_pYno:b-mwe9ow) ([raw file](https://github.com/cockroachdb/cockroach/blob/032112361de72739e4485dee6f235fed7b43eb19/pkg/ui/src/redux/alerts.ts#L46)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

It looks like you updated the comment to say "expanded".  My point above remains, I think this would be cleaner if the logic were reversed and "collapsed" were the stored value...
</blockquote></details>

Gonna leave it be I think.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/cockroach/23516#-:-L8F1RZhAxHFZV0qoTX1:b-imm9a0)*
<!-- Sent from Reviewable.io -->

couchand commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> modulo copy review and design review.

I agree with Amruta's comment about the screenshot, and I'd add that it's weird to have OS-specific chrome in the screenshot.

---

Reviewed 5 of 7 files at r11, 2 of 3 files at r12, 2 of 2 files at r13.
Review status: all files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

*[pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx, line 91 at r13](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8IwwfqlaSOdCTprWSE:-L8IwwfqlaSOdCTprWSF:b-y78f2g) ([raw file](https://github.com/cockroachdb/cockroach/blob/177c2208713e00326e15a65d1b3633da66cf3131/pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx#L91)):*
> ```XML
>         return this.renderExpanded();
>       } catch (e) {
>         console.error(e);
> ```

I'm assuming this was for debugging purposes?  It's really annoying that somewhere in our usage of React all thrown errors are being caught and ignored...  We need to figure that out.

---

*[pkg/ui/src/util/localities.spec.ts, line 15 at r13](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-L8IxqLJeFEP7U_heTRr:-L8IxqLJeFEP7U_heTRs:byeu5mz) ([raw file](https://github.com/cockroachdb/cockroach/blob/177c2208713e00326e15a65d1b3633da66cf3131/pkg/ui/src/util/localities.spec.ts#L15)):*
> ```XML
>   allNodesHaveLocality,
> } from "./localities";
> import { cockroach } from "oss/src/js/protos";
> ```

hahahaha oss :-( whatever happened to that linter you started writing?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516#-:-L8Iy29Ze0LLlh3YiqYO:b77b0to)*
<!-- Sent from Reviewable.io -->

Commit:
ui: clusterviz: tweak copy in instructions box & gate page

Release note: None

vilterp commented:
K, tweaked copy to @Amruta-Ranade's suggestions. Haven't change the screenshot though. Let's all sign off and merge this Monday. (cc @josueeee)

Flipped the setting back to `collapsed` since I was bitten by this overriding what was set: https://github.com/cockroachdb/cockroach/blob/230e843a957f7ef984ea21ff901216aabf876322/pkg/ui/src/redux/localsettings.ts#L99

---

Review status: 3 of 11 files reviewed at latest revision, 3 unresolved discussions, some commit checks pending.

---

*[pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx, line 91 at r13](https://beta.reviewable.io/reviews/cockroachdb/cockroach/23516#-L8IwwfqlaSOdCTprWSE:-L8J98Fb54orHfMjmyKW:bwh44ek) ([raw file](https://github.com/cockroachdb/cockroach/blob/177c2208713e00326e15a65d1b3633da66cf3131/pkg/ui/ccl/src/views/clusterviz/components/instructionsBox/index.tsx#L91)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

I'm assuming this was for debugging purposes?  It's really annoying that somewhere in our usage of React all thrown errors are being caught and ignored...  We need to figure that out.
</blockquote></details>

Whoops, removed. Yeah, the errors being swallowed drives me crazy. Especially since when I open the inspector it freezes for 10 seconds while it renders vendor.dll.js.

Maybe React 16 componentDidCatch will help? Or just a try/catch somewhere else? Idk.

---

*[pkg/ui/src/util/localities.spec.ts, line 15 at r13](https://beta.reviewable.io/reviews/cockroachdb/cockroach/23516#-L8IxqLJeFEP7U_heTRr:-L8J9R6XDVb_49fXmNEu:b-914860) ([raw file](https://github.com/cockroachdb/cockroach/blob/177c2208713e00326e15a65d1b3633da66cf3131/pkg/ui/src/util/localities.spec.ts#L15)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

hahahaha oss :-( whatever happened to that linter you started writing?
</blockquote></details>

Gah. Lost in a branch somewhere — however, after writing the Go program, I realized it's equivalent to just `git grep` with a couple of exceptions — should be possible with a little bash-fu.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/cockroach/23516#-:-L8JPo37D7viGt6bf17d:bde4vjk)*
<!-- Sent from Reviewable.io -->

couchand commented:
Reviewed 4 of 8 files at r14, 2 of 2 files at r16, 2 of 2 files at r17.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/23516)*
<!-- Sent from Reviewable.io -->

josueeee commented:
Design updates:

- [ ] PIP progress indication clarity 
<img width="1184" alt="imga" src="https://user-images.githubusercontent.com/29442020/37978492-27b251f0-31b4-11e8-9494-e7077a992d36.png">

- [ ] Collapsed PIP icon scale 
<img width="306" alt="imgb" src="https://user-images.githubusercontent.com/29442020/37978509-33bf419c-31b4-11e8-8af8-d7bb2f73d259.png">

- [ ] Enterprise gate page - text line height 
<img width="2313" alt="imgc" src="https://user-images.githubusercontent.com/29442020/37978517-37802922-31b4-11e8-855a-aa1dc5b303b9.png">

- [ ] Update Node Map screenshots with new map colors 
<img width="1260" alt="imgd" src="https://user-images.githubusercontent.com/29442020/37978525-3e1149e2-31b4-11e8-82a3-0bfe5b30b895.png">

- [ ] Enterprise gate page - update step 1 screenshot with the new screenshot 
<img width="986" alt="imge" src="https://user-images.githubusercontent.com/29442020/37978537-47ea22d6-31b4-11e8-8a3c-7769e44842c0.png">

couchand committed:
ui: show checkmark for completed clusterviz instructions items

Release note: None

couchand committed:
ui: clusterviz sizing tweaks

Release note: None

couchand committed:
ui: fix scroll issue on license instructions page

Release note: None

couchand commented:
bors ping

couchand commented:
bors try

couchand commented:
that's what i wanted to see ^^^

couchand committed:
ui: update node map instructions images

Release note: None

vilterp commented:
Tried it locally. LGTM!
